storage/innobase/log/log0recv.cc

```diff
@@ -2668,9 +2667,8 @@ ulint recv_parse_log_rec(mlog_id_t *type, byte *ptr, byte *end_ptr,
           mlog_parse_initial_dict_log_record(ptr, end_ptr, type, &id, &version);
 
       if (new_ptr != nullptr) {
-        new_ptr =
-            (apply ? recv_sys->metadata_recover : log_online_metadata_recover)
-                ->parseMetadataLog(id, version, new_ptr, end_ptr, apply);
+        new_ptr = recv_sys->metadata_recover->parseMetadataLog(
```

Comment:
This is dangerous. When we do log online parsing, recv_sys is nullptr. Recipe for  :boom:

Reply:
But apply before the change was always true, so nothing changes here in execution. Also in upstream they do not check for apply (because they do not have it) and go straight to recv_sys->metadata_recover->parseMetadataLog.

Reply:
Ok, this is an existing issue. ie. During log online parsing, which happens during the server, we should use log_online_metadata_recover. Because recv objects are freed after the recovery. So we definitely need to use log_online_metadata_recover.

Please create a new bug to handle this. I know that log online parsing is broken in 8.0 at least just enabling it, shouldn't the crash the server.

Reply:
I was wrong here too. This is indeed needed. However I changed the apply variable to online_log - to indicate that we are parsing online log and not applying anything. This confused me.